client/src/components/Home/components/Editor.vue

```diff
@@ -1,56 +1,85 @@
 <template>
   <div
     class="editor"
-    @keydown.prevent="handleKeyboardPress"
-    tabindex="0"
     data-test="editor"
-  >{{documentData}}</div>
+    @keydown="handleKeyboardPress"
+    @keyup="handleKeyboardPress"
+    tabindex="0"
+  >
+    <div
+      class="row"
+      data-test="editor-row"
+      v-for="(row, index) in documentData"
+      :class="[`${(documentData.length - 1) === index ? 'current' : ''}`]"
+      v-bind:key=index
+    >
+      <div class="row-number" data-test="row-number">{{ index }}</div>
+      <div data-test="row-data">{{ row }}</div>
+    </div>
+  </div>
 </template>
 
 <script>
-const getKeyValueToRender = (keyCode, keyValue) => {
-  switch (keyCode) {
-    // Enter
-    case 13:
-      return '\n';
-    // Tab
-    case 9:
-      return '\xa0\xa0';
-    default:
-      return keyValue;
-  }
-};
+import {
+  getDefaultValue,
+  listOfSpecialKeys,
+  setValueToStorage,
+  updateDocumentData,
+} from './utilities';
 
-const getDefaultValue = () => {
-  if (window && window.localStorage) {
-    return window.localStorage.getItem('documentData') || '';
-  }
-  return '';
-};
+const KEY_DOWN_EVENT = 'keydown';
+const KEY_UP_EVENT = 'keyup';
 
-const setValueToStorage = (value) => {
-  if (window && window.localStorage) {
-    window.localStorage.setItem('documentData', value);
-  }
-};
+const keysPressed = (() => {
+  let currentShortcutKeysPressed = [];
+  const addEventToRegister = ({ keyCode, type }) => {
+    const hasProperty = Object.prototype.hasOwnProperty.call(
+      listOfSpecialKeys,
+      keyCode,
+    );
+    if (hasProperty) {
+      if (type === KEY_DOWN_EVENT) {
+        currentShortcutKeysPressed = [...currentShortcutKeysPressed, keyCode];
+      } else {
+        currentShortcutKeysPressed = currentShortcutKeysPressed.filter(
+          (item) => item === keyCode,
+        );
+      }
+    }
+  };
+  const shortcutKeyIsBeingPressed = () => currentShortcutKeysPressed.length > 0;
+  const isShift = ({ keyCode }) => keyCode === 16;
```

Comment:
The `keydown` event also has boolean properties (`.ctrlKey`, `.shiftKey`, `.altKey`, and `.metaKey`) that return true if the control, shift, alt, and meta keys, respectively, are being held down while a key is pressed. It might shorten our code if we made use of those properties rather than define our own functions. That is, unless we're planning on more advanced key combinations. Then the function you've written seems like it could handle such situations more elegantly.